### PR TITLE
Fixed wrong order of input, unused import and changed name of protocol

### PR DESCRIPTION
**Pull request by NateZ7**
Latitude should be before Longitude. Changed the name of the protocol. unused import removed.

### CODE REVIEW

**Comment by tananaev** [comment]
Why are we renaming the protocol?

**Comment by slobglob** [comment]
@tananaev the original name `globekeeper` is not right here, this is for a custom protocol and the correct name is the one here.

**Comment by tananaev** [comment]
Name of classes should be in camel-case like `AustinNbXxx`.

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java [hunk 1]
```diff
@@ -57,11 +63,16 @@ protected Object decode(
...
 
         position.setValid(true);
-        position.setLongitude(Double.parseDouble(parser.next().replace(',', '.')));
         position.setLatitude(Double.parseDouble(parser.next().replace(',', '.')));
+        position.setLongitude(Double.parseDouble(parser.next().replace(',', '.')));
+        position.set("azimuth", parser.nextInt());
+        position.set("angle", parser.nextInt());
+        position.set("range", parser.nextInt());
+        position.set("out_of_range", parser.nextInt());
+        position.set("currier", parser.next());
```
Comment (tananaev) [inline src/org/traccar/protocol/AustinNbProtocolDecoder.java]: What is "currier"?

https://en.wikipedia.org/wiki/Currier

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java [hunk 2]
```diff
@@ -57,11 +63,16 @@ protected Object decode(
...
+        position.setTime(parser.nextDateTime(Parser.DateTimeFormat.YMD_HMS, TimeZone.getDefault().getID()));
 
         position.setValid(true);
-        position.setLongitude(Double.parseDouble(parser.next().replace(',', '.')));
         position.setLatitude(Double.parseDouble(parser.next().replace(',', '.')));
+        position.setLongitude(Double.parseDouble(parser.next().replace(',', '.')));
+        position.set("azimuth", parser.nextInt());
+        position.set("angle", parser.nextInt());
+        position.set("range", parser.nextInt());
+        position.set("out_of_range", parser.nextInt());
```
Comment (tananaev) [inline src/org/traccar/protocol/AustinNbProtocolDecoder.java]: This doesn't comply with naming convention. Also, what does it mean?

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java [hunk 3]
```diff
@@ -57,11 +63,16 @@ protected Object decode(
...
         position.setDeviceId(deviceSession.getDeviceId());
 
-        position.setTime(parser.nextDateTime());
+        position.setTime(parser.nextDateTime(Parser.DateTimeFormat.YMD_HMS, TimeZone.getDefault().getID()));
 
         position.setValid(true);
-        position.setLongitude(Double.parseDouble(parser.next().replace(',', '.')));
         position.setLatitude(Double.parseDouble(parser.next().replace(',', '.')));
+        position.setLongitude(Double.parseDouble(parser.next().replace(',', '.')));
+        position.set("azimuth", parser.nextInt());
```
Comment (tananaev) [inline src/org/traccar/protocol/AustinNbProtocolDecoder.java]: What's this? Why is it not assigned to course properly?

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java [hunk 4]
```diff
@@ -57,11 +63,16 @@ protected Object decode(
...
+        position.set("angle", parser.nextInt());
```
Comment (tananaev) [inline src/org/traccar/protocol/AustinNbProtocolDecoder.java]: What's this?

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java [hunk 5]
```diff
@@ -57,11 +63,16 @@ protected Object decode(
...
+        position.set("range", parser.nextInt());
```
Comment (tananaev) [inline src/org/traccar/protocol/AustinNbProtocolDecoder.java]: What's this one as well?

File: test/org/traccar/protocol/AustinNbProtocolDecoderTest.java [hunk 6]
```diff
@@ -3,30 +3,30 @@
...
 
     @Test
     public void testDecode() throws Exception {
 
-        GlobeKeeperProtocolDecoder decoder = new GlobeKeeperProtocolDecoder(new GlobeKeeperProtocol());
+        AustinNbProtocolDecoder decoder = new AustinNbProtocolDecoder(new AustinNbProtocol());
 
         verifyPosition(decoder, text(
-                "48666666666;2017-01-01 16:31:01;52,1133308410645;21,1000003814697;310;120;2292;1"));
+                "48666666666;2017-01-01 16:31:01;52,1133308410645;21,1000003814697;310;120;2292;1;ORANGE"));
```
Comment (tananaev) [inline test/org/traccar/protocol/AustinNbProtocolDecoderTest.java]: Are you changing protocol format?

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java [hunk 7]
```diff
@@ -37,6 +38,11 @@ public GlobeKeeperProtocolDecoder(GlobeKeeperProtocol protocol) {
             .number("(dd):(dd):(dd);")           // time
             .number("(-?d+,d+);")                // latitude
             .number("(-?d+,d+);")                // longitude
+            .number("(d+);")                     // azimuth
+            .number("(d+);")                     // angle
+            .number("(d+);")                     // range
+            .number("(d+);")                     // out of range
+            .expression("(.*)")                          // operator
```
Comment (tananaev) [inline src/org/traccar/protocol/AustinNbProtocolDecoder.java]: Indentation of the comment is wrong.

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java (hunk 1 above)
Comment (NateZ7) [reply src/org/traccar/protocol/AustinNbProtocolDecoder.java]: An extra info we need for the protocol.

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java (hunk 2 above)
Comment (NateZ7) [reply src/org/traccar/protocol/AustinNbProtocolDecoder.java]: An extra info we need for the protocol.
I will name by the convention.

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java (hunk 3 above)
Comment (NateZ7) [reply src/org/traccar/protocol/AustinNbProtocolDecoder.java]: An extra info we need for the protocol.

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java (hunk 4 above)
Comment (NateZ7) [reply src/org/traccar/protocol/AustinNbProtocolDecoder.java]: An extra info we need for the protocol.

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java (hunk 5 above)
Comment (NateZ7) [reply src/org/traccar/protocol/AustinNbProtocolDecoder.java]: An extra info we need for the protocol.

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java (hunk 7 above)
Comment (NateZ7) [reply src/org/traccar/protocol/AustinNbProtocolDecoder.java]: Will fix.

**Comment by tananaev** [comment]
If you want to include extra information in the official Traccar branch you would need to explain what each field means.

**Comment by NateZ7** [comment]
Sure,

How would you like me to do so ? comments inside the code is fine ? if
otherwise just let me know

**Comment by tananaev** [comment]
Comments on GitHub.

File: test/org/traccar/protocol/AustinNbProtocolDecoderTest.java (hunk 6 above)
Comment (NateZ7) [reply test/org/traccar/protocol/AustinNbProtocolDecoderTest.java]: Yes

**Comment by NateZ7** [comment]
Thanks, Updated comments on Github.

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java (hunk 3 above)
Comment (tananaev) [reply src/org/traccar/protocol/AustinNbProtocolDecoder.java]: This is exactly what course is. It's a properly in position model. You should use it instead.

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java (hunk 4 above)
Comment (tananaev) [reply src/org/traccar/protocol/AustinNbProtocolDecoder.java]: OK, we can probably keep it as it is.

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java (hunk 5 above)
Comment (tananaev) [reply src/org/traccar/protocol/AustinNbProtocolDecoder.java]: OK, we can probably keep it as it is.

File: src/org/traccar/protocol/AustinNbProtocolDecoder.java [hunk 8]
```diff
@@ -57,11 +63,16 @@ protected Object decode(
...
 
         position.setValid(true);
-        position.setLongitude(Double.parseDouble(parser.next().replace(',', '.')));
         position.setLatitude(Double.parseDouble(parser.next().replace(',', '.')));
+        position.setLongitude(Double.parseDouble(parser.next().replace(',', '.')));
+        position.set("azimuth", parser.nextInt());
+        position.set("angle", parser.nextInt());
+        position.set("range", parser.nextInt());
+        position.set("outOfRange", parser.nextInt());
+        position.set("currier", parser.next());
```
Comment (tananaev) [inline src/org/traccar/protocol/AustinNbProtocolDecoder.java]: I believe this is still wrong.

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**